src/Sentry/Protocol/Metrics/Metric.cs

```diff
@@ -0,0 +1,119 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+using ISentrySerializable = Sentry.Protocol.Envelopes.ISerializable;
+
+namespace Sentry.Protocol.Metrics;
+
+internal abstract class Metric : IJsonSerializable, ISentrySerializable
+{
+    protected Metric() : this(string.Empty)
+    {
+    }
+
+    protected Metric(string key, MeasurementUnit? unit = null, IDictionary<string, string>? tags = null, DateTime? timestamp = null)
+    {
+        Key = key;
+        Unit = unit;
+        Tags = tags ?? new Dictionary<string, string>();
```

Comment:
can we set the tags lazily instead? I imagine a lots of metrics will have no tags so we can spare the dic allocations